core/primitives/src/views.rs

```diff
@@ -207,11 +207,28 @@ pub struct StateItem {
     pub proof: TrieProofPath,
 }
 
+/// A proof can be present or absent both on a result of a function but also on [ViewStateResult]. This
```

Comment:
This isn’t whether proof is present or not.  This is whether the value is present or not.  Perhaps just call it `Presence`.  I also don’t like `From<bool>` implementation.